OpenAI.SDK/ObjectModels/RequestModels/CompletionCreateRequest.cs

```diff
@@ -18,9 +18,38 @@ public record CompletionCreateRequest : IModelValidate, IOpenAiModels.ITemperatu
         ///     the model will generate as if from the beginning of a new document.
         /// </summary>
         /// <see cref="https://beta.openai.com/docs/api-reference/completions/create#completions/create-prompt" />
-        [JsonPropertyName("prompt")]
+        [JsonIgnore]
         public string? Prompt { get; set; }
 
+        /// <summary>
+        ///     The prompt(s) to generate completions for, encoded as a string, a list of strings, or a list of token lists.
+        ///     Note that endoftext is the document separator that the model sees during training, so if a prompt is not specified
+        ///     the model will generate as if from the beginning of a new document.
+        /// </summary>
+        /// <see cref="https://beta.openai.com/docs/api-reference/completions/create#completions/create-prompt" />
+        [JsonIgnore]
```

Comment:
Personally I think it would be better to use generics for this kind of thing. It makes the intent much more clear upfront when consuming the it rather than an exception being thrown later down the line. Either way the caller needs to change something so rather use the type to enforce this constraint.

e.g

public class TextCompletionRequestBase<T>
{ 
        public T Prompt { get; set; }
}

public class TextCompletionRequest : TextCompletionRequestBase<string>
  {
      public TextCompletionRequest(string model, string prompt)
      {
          Model = model;
          Prompt = prompt;
      }
  }

  public class TextCompletionListRequest: TextCompletionRequestBase<IList<string>>
  {
      public TextCompletionListRequest(string model, IList<string> prompt)
      {
          Model = model;
          Prompt = prompt;
      }
  }


I just done this too as it was something I need to implement, Feel free to take a look here 
https://github.com/jodendaal/OpenAI.Net/pull/15/commits/7f1e7d060b0ced2f731f89bf33222b2fb91d28e2#diff-65c897d1314d05807c5a5bf53e8e1f6a3208bfa06ba21ee41f764e7346c02d57

Reply:
This way we would end up with 2 different classes and also more breaking change. I prefer to keep things as simple as much. Another thing, imagine that you wanted to use `prompt` as a string and `stop` as a List. To support this you will need 4 different classes. If OpenAI introduces new property now we will need 8.

I thought about introducing a new class type that allows assigning string or sting lists but I think it wouldn't be worth it because of the reason I mentioned above. 

Also, the initial calculation method was merging the prompt and prompt list in a new string list instead of throwing an exception, but when I think about it I couldn't find any use case. If any developer uses both prompt and prompt list probably doing it intentionally. That's why I decided the throw a new exception. 

I can change the implementation if more people disagree with me and address solutions to my concerns.

Reply:
That's a fair point, I didn't realize the stop was also either an array or string.  For existing support you'd have to keep it as you have done.